app/assets/javascripts/utils/course_utils.js

```diff
@@ -92,18 +92,12 @@ const CourseUtils = class {
     const language = assignment.language || 'en';
     const project = assignment.project || 'wikipedia';
     const articleUrl = assignment.article_url || this.urlFromTitleAndWiki(assignment.article_title, language, project);
-    const formattedTitle = this.formattedArticleTitle(
```

Comment:
Instead of removing this and setting formatted_title in other places every time this is called, you could do article.formatted_title  = this.formattedArticleTitle(article) before the return.

If necessary, you could also make articleFromAssignment take an optional homeWiki argument to include.

In general, the more shared logic we can move into CourseUtils, the better. Isolating that logic into the shared vanilla Javascript function makes it much easier to test and maintain.